### DISCUSSION TITLE
CAFE analysis on trees

### DISCUSSION THREAD

**Discussion by beantkapoor786** [hahnlab/CAFE5#91] [Q&A] [answered]
Hello, thank you for this wonderful tool. I am trying to run CAFE on 10 tree genomes (eudicots). I have generated a list of orthogroups using OrthoFinder. There are 26,599 orthogroups. There were some orthogroups with high gene variance so I used the python script `clade_and_size_filter.py`. It removed only 6 orthogroups. I also created the rooted, ultrametric tree in newick format. However, every time I run CAFE it gave me an error saying `Families with largest size differentials:`. I have tried removing all those families to the extent that the output now looks like this -
```
Filtering the number of families from: 5615 ==> 628

Optimizer strategy: Nelder-Mead with similarity cutoff

Iterations: 300
Expansion: 2
Reflection: 1

Starting Search for Initial Parameter Values


Score (-lnL):             inf
Lambda: 0.002007918912153
Score (-lnL):             inf
Lambda: 0.0019472643570216
Score (-lnL):             inf
Lambda: 0.0023958458617025
Score (-lnL):             inf
Lambda: 0.0021243542790506
Score (-lnL):             inf
Lambda: 0.0023637872314451
Score (-lnL):             inf
Lambda: 0.0021610538497862
Score (-lnL):             inf
Lambda: 0.0018825611056004
Score (-lnL):             inf
Lambda: 0.0019894551505632
Score (-lnL):             inf
Lambda: 0.001558168157784
Score (-lnL):             inf
Lambda: 0.0021792220207378
Score (-lnL):             inf
Lambda: 0.0017925221514194
Score (-lnL):             inf
Lambda: 0.0023655383510853
Score (-lnL):             inf

Families with largest size differentials:
OG0017332: 4
OG0003991: 4
OG0003780: 4
OG0003790: 4
OG0003788: 4
OG0003786: 4
OG0004444: 3
OG0004006: 3
OG0004003: 3
OG0004440: 3
OG0004001: 3
OG0004000: 3
OG0004442: 3
OG0004012: 3
OG0004445: 3
OG0003993: 3
OG0003989: 3
OG0003988: 3
OG0004450: 3
OG0004435: 3

You may want to try removing the top few families with the largest difference
between the max and min counts and then re-run the analysis.
```
I have gone from 26,599 orthogroups to just 5,615 and CAFE is still not able to generate a reasonable lamba value. Can you please suggest something? Any help would be greatly appreciated!  

Orthogroups gene count file after using python script - 
[Orthogroups.GeneCount.filtered.txt](https://github.com/hahnlab/CAFE5/files/9931308/Orthogroups.GeneCount.filtered.txt)

Filtered down to 5,615 orthogroups - 
[filtered_gene_families.txt](https://github.com/hahnlab/CAFE5/files/9931553/filtered_gene_families.txt)

Ultrametric tree file - 
[ultrametric_tree.txt](https://github.com/hahnlab/CAFE5/files/9931315/ultrametric_tree.txt)

**Comment by benfulton** [comment, marked as answer]
Your branch for Athalania has a length of 904. CAFE expects the tree values to be roughly representative of millions of years, so the branch is something like the entire history of plants :) Experiment by dividing all the values in the tree by 10 or 100 and you should be able to at least get a result.

**Reply by beantkapoor786** [reply]
Thank you for the suggestion, it worked. However, how ideal is it to divide the branch lengths by 10 or 100? Should I try to find a better list of species for the analysis?

**Reply by benfulton** [reply]
Mathematically it's fine, keeping in mind that the result lambda is 1/10th or 1/100th of the real value. I cannot guess whether it's appropriate biologically - that's something for you and your adviser to decide :)

**Reply by LiS-88** [reply]
> 谢谢你的建议，很有效。然而，将分支长度除以10或100有多理想？我应该试着找到一个更好的物种列表来进行分析吗？

hello~
How did you solve the problem? Did you adjust the parameters? Or did you adjust the file that was entered into cafe5?

**Reply by beantkapoor786** [reply]
Hi, I adjusted the branch lengths so that they look like they are divided by 10. After that, I was able to run cafe.

**Reply by beantkapoor786** [reply]
> Mathematically it's fine, keeping in mind that the result lambda is 1/10th or 1/100th of the real value. I cannot guess whether it's appropriate biologically - that's something for you and your adviser to decide :)

Okay, thank you for the suggestion.
